distribution/tools/geoip-cli/src/main/java/org/elasticsearch/geoip/GeoIpCli.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.geoip;
+
+import joptsimple.OptionSet;
+import joptsimple.OptionSpec;
+import org.elasticsearch.cli.Command;
+import org.elasticsearch.cli.Terminal;
+import org.elasticsearch.common.SuppressForbidden;
+import org.elasticsearch.common.hash.MessageDigests;
+import org.elasticsearch.common.io.PathUtils;
+import org.elasticsearch.common.xcontent.XContentGenerator;
+import org.elasticsearch.common.xcontent.XContentType;
+
+import java.io.BufferedInputStream;
+import java.io.BufferedOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.security.DigestInputStream;
+import java.security.MessageDigest;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Locale;
+import java.util.stream.Collectors;
+import java.util.zip.GZIPOutputStream;
+
+import static java.nio.file.StandardOpenOption.CREATE;
+import static java.nio.file.StandardOpenOption.TRUNCATE_EXISTING;
+
+public class GeoIpCli extends Command {
+
+    private static final byte[] EMPTY_BUF = new byte[512];
+    private final OptionSpec<String> directoryArg;
+
+    public GeoIpCli() {
+        super("A CLI tool to prepare local GeoIp database service", () -> {});
+        directoryArg = parser.acceptsAll(Arrays.asList("d", "directory"), "Directory to process").withRequiredArg().defaultsTo(".");
+    }
+
+    @Override
+    protected void execute(Terminal terminal, OptionSet options) throws Exception {
+        Path directory = getPath(options.valueOf(directoryArg));
+        packDatabasesToTgz(terminal, directory);
+        createOverviewJson(terminal, directory);
+    }
+
+    @SuppressForbidden(reason = "file arg for cli")
+    private Path getPath(String file) {
+        return PathUtils.get(file);
+    }
+
+    private void packDatabasesToTgz(Terminal terminal, Path directory) throws IOException {
+        List<Path> toPack = Files.list(directory).filter(p -> p.getFileName().toString().endsWith(".mmdb")).collect(Collectors.toList());
+        for (Path path : toPack) {
+            String fileName = path.getFileName().toString();
+            Path compressedPath = path.resolveSibling(fileName.replaceAll("mmdb$", "") + "tgz");
```

Comment:
Should we allow for an additional argument that allows specify the target directory (which then defaults to the source directory)? The `d` argument is now both the source and target directory.

Reply:
Yes, this is good idea, I've changed `d[ir]` to `s[ource]` and `t[arget]` pair and added tests for using 2 different dirs and defaulting to source